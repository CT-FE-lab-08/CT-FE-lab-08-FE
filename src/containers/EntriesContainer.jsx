import React from 'react';
import NewEntryControls from '../components/app/controls/NewEntryControls';
import EntriesList from '../components/app/entries/EntriesList';

export default function EntriesContainer() {
  return (
    <div>
      Entries Container
      <NewEntryControls />
      <EntriesList />
    </div>
  );
}
