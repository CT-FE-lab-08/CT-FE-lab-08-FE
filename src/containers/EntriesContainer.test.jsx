/* eslint-disable max-len */
import React from 'react';
import { MemoryRouter } from 'react-router';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import userEvent from '@testing-library/user-event';
import listData from '../components/fixtures/listData.json';
import EntriesContainer from './EntriesContainer';

const server = setupServer(
  rest.get('https://still-journey-26608.herokuapp.com/api/v1/alchemy-cry-lab',
    (req, res, ctx) => {
      return res(
        ctx.status(200),
        ctx.json(listData)
      );
    }),

  rest.post('https://still-journey-26608.herokuapp.com/api/v1/alchemy-cry-lab',
    (req, res, ctx) => {
      return res(
        ctx.status(200),
        ctx.json({
          date: '2021-09-27', 
          event: true, 
          id: '24', 
          name: 'test test',
          note: 'aaaaaah'
        }));
    }
  ));

describe('Entries Container Test', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('should enter a new cry and re-render the DOM with the new entry', async () => {
  
    render(
      <MemoryRouter>
        <EntriesContainer />
      </MemoryRouter>
    );

    screen.getByText('Loading...');

    const nameInput = await screen.findByPlaceholderText('your name');
    const noteInput = await screen.findByPlaceholderText('tell me about it');
    const submitButton = await screen.findByRole(
      'button',
      { name: 'submit' }
    );
      
    userEvent.type(nameInput, 'test test');
    userEvent.type(noteInput, 'aaaaaah');

    act(() => {
      fireEvent.click(submitButton);
    });

    return waitFor(() => {
      const response = screen.getByText(
        'Your entry was successfully posted. If you think you\'ll want to modify this entry in the future, store this id to access the entry:'
      );
  
      expect(response).toBeInTheDocument();

    });
  });
});
