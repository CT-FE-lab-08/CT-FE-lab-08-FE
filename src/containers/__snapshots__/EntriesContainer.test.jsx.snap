// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entries Container Test should enter a new cry and re-render the DOM with the new entry 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <p>
            Your entry was successfully posted. If you think you'll want to modify this entry in the future, store this id to access the entry:
          </p>
          <p>
            24
          </p>
        </div>
        <h2>
          Entries
        </h2>
        <ul>
          <li>
            <p>
              test test didn't cry on 2021-09-27. "aaaaaah"
            </p>
          </li>
          <li>
            <div>
              <p>
                asdf cried on 2021-09-27. "asdf"
              </p>
            </div>
          </li>
          <li>
            <div>
              <p>
                cha cha cha cried on 2021-09-27. "woooooooork it"
              </p>
            </div>
          </li>
          <li>
            <div>
              <p>
                blah blah cried on 2021-09-27. "sdf"
              </p>
            </div>
          </li>
          <li>
            <p>
              please work didn't cry on 2021-09-27. "jafhgkjah"
            </p>
          </li>
          <li>
            <p>
              stuff didn't cry on 2021-09-27. "stuff stuff"
            </p>
          </li>
          <li>
            <p>
              tessssst didn't cry on 2021-09-27. "hi"
            </p>
          </li>
          <li>
            <div>
              <p>
                test3 cried on 2021-09-27. "test3"
              </p>
            </div>
          </li>
          <li>
            <p>
              test2 didn't cry on 2021-09-27. "test2"
            </p>
          </li>
          <li>
            <div>
              <p>
                DJ cried on 2021-09-27. "my sorrow is an ocean"
              </p>
            </div>
          </li>
        </ul>
        <a
          href="/edit"
        >
          <p>
            Edit an existing entry
          </p>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <p>
          Your entry was successfully posted. If you think you'll want to modify this entry in the future, store this id to access the entry:
        </p>
        <p>
          24
        </p>
      </div>
      <h2>
        Entries
      </h2>
      <ul>
        <li>
          <p>
            test test didn't cry on 2021-09-27. "aaaaaah"
          </p>
        </li>
        <li>
          <div>
            <p>
              asdf cried on 2021-09-27. "asdf"
            </p>
          </div>
        </li>
        <li>
          <div>
            <p>
              cha cha cha cried on 2021-09-27. "woooooooork it"
            </p>
          </div>
        </li>
        <li>
          <div>
            <p>
              blah blah cried on 2021-09-27. "sdf"
            </p>
          </div>
        </li>
        <li>
          <p>
            please work didn't cry on 2021-09-27. "jafhgkjah"
          </p>
        </li>
        <li>
          <p>
            stuff didn't cry on 2021-09-27. "stuff stuff"
          </p>
        </li>
        <li>
          <p>
            tessssst didn't cry on 2021-09-27. "hi"
          </p>
        </li>
        <li>
          <div>
            <p>
              test3 cried on 2021-09-27. "test3"
            </p>
          </div>
        </li>
        <li>
          <p>
            test2 didn't cry on 2021-09-27. "test2"
          </p>
        </li>
        <li>
          <div>
            <p>
              DJ cried on 2021-09-27. "my sorrow is an ocean"
            </p>
          </div>
        </li>
      </ul>
      <a
        href="/edit"
      >
        <p>
          Edit an existing entry
        </p>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
